tests/aws/services/events/test_events.py

```diff
@@ -1742,6 +1742,293 @@ def test_process_pattern_to_single_matching_rules_single_target(
+        for rule_name in rule_names + [other_rule]:
+            try:
+                # Remove targets first (only for rules that have targets)
+                if rule_name in rule_names:  # other_rule has no targets
+                    aws_client.events.remove_targets(
+                        Rule=rule_name,
+                        EventBusName=bus_name,
+                        Ids=[f"target-{rule_names.index(rule_name)}"],
+                    )
+                # Delete rule
+                aws_client.events.delete_rule(
+                    Name=rule_name,
+                    EventBusName=bus_name,
+                    Force=True,
+                )
+            except aws_client.events.exceptions.ResourceNotFoundException:
+                # Ignore if rule or event bus was already deleted
+                pass
+
+        # Delete custom event bus if we created one
+        if bus_name != "default":
+            try:
+                aws_client.events.delete_event_bus(Name=bus_name)
+            except aws_client.events.exceptions.ResourceNotFoundException:
+                # Ignore if event bus was already deleted
+                pass
+
+    @markers.aws.validated
+    @pytest.mark.parametrize("bus_name", ["custom", "default"])
+    def test_list_rule_names_by_target_with_limit(
+        self,
+        bus_name,
+        events_create_event_bus,
+        events_put_rule,
+        sqs_create_queue,
+        sqs_get_queue_arn,
+        aws_client,
+        snapshot,
+        clean_up,
+    ):
+        """Test the ListRuleNamesByTarget API with pagination to verify it correctly handles limits and next tokens."""
+        # Create an SQS queue to use as a target
+        queue_name = f"queue-{short_uid()}"
+        queue_url = sqs_create_queue(QueueName=queue_name)
+        queue_arn = sqs_get_queue_arn(queue_url)
+        snapshot.add_transformer(snapshot.transform.regex(queue_arn, "<queue-arn>"))
+
+        # Create an event bus if using custom bus
+        if bus_name == "custom":
+            bus_name = f"bus-{short_uid()}"
+            events_create_event_bus(Name=bus_name)
+            snapshot.add_transformer(snapshot.transform.regex(bus_name, "<bus-name>"))
+
+        # Create multiple rules targeting the same SQS queue
+        rule_prefix = f"rule-{short_uid()}-"
+        snapshot.add_transformer(snapshot.transform.regex(rule_prefix, "<rule-prefix>"))
+        rule_names = []
+
+        # Create 5 rules all targeting the same SQS queue
+        for i in range(5):
+            rule_name = f"{rule_prefix}{i}"
+            rule_names.append(rule_name)
+            events_put_rule(
+                Name=rule_name,
+                EventBusName=bus_name,
+                EventPattern=json.dumps({"source": [f"source-{i}"]}),
+            )
+
+            # Add the SQS queue as a target for this rule
+            target_id = f"target-{i}"
+            snapshot.add_transformer(snapshot.transform.regex(target_id, f"<target-id-{i}>"))
```

Comment:
Like with the SQS ARN and EventBus name, we don't capture target ids in the response
```suggestion
```